library/src/main/java/com/felipecsl/gifimageview/library/GifImageView.java

```diff
@@ -151,8 +155,15 @@ public int getGifHeight() {
         }
         gifDecoder.advance();
         try {
-          Thread
-              .sleep(framesDisplayDuration > 0 ? framesDisplayDuration : gifDecoder.getNextDelay());
+          int delay = gifDecoder.getNextDelay();
+          // Sleep for frame duration minus time already spent on frame decode
+          // Actually we need next frame decode duration here,
+          // but I use previous frame time to make code more readable
+          delay -= frame_decode_time;
```

Comment:
if frame decode time is bigger than the delay, it could end up being a negative, no?